core/src/main/java/hudson/ProxyConfiguration.java

```diff
@@ -259,6 +259,40 @@ public PasswordAuthentication getPasswordAuthentication() {
         return con;
     }
 
+    /**
+     * This method should be used wherever {@link URL#openConnection()} to internet URLs is invoked directly and a redirection is possible.
+     * @since TODO
+     */
+    public static URLConnection openAllowsRedirection(URL url) throws IOException {
```

Comment:
`openAllowsRedirection` and `setInstanceFollowRedirects` are too close in name given that this method is not about the latter.

Reply:
Maybe `openURLAndRedirect` combined with the change of the comment?

Reply:
Changed